pandas/tseries/tests/test_timeseries.py

```diff
@@ -4911,6 +4968,63 @@ def test_partial_slice_hourly(self):
         self.assertEqual(s['2005-1-1 20:00'], s.ix[0])
         self.assertRaises(Exception, s.__getitem__, '2004-12-31 00:15')
 
+        # GH14856
+        # DatetimeIndex without explicit freq
+        df = DataFrame({'a': [1, 2, 3]}, DatetimeIndex(['2011-12-31 23',
+                                                        '2012-01-01 00',
+                                                        '2012-01-01 01']),
+                       dtype=np.int64)
+
```

Comment:
you are repeating lots of tests with only a slight variation

can u so it instead with a loop